### PR TITLE
tests(clustering): sync v1 should work with or without privileged agent

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Refactored some code in #13857

KAG-5807
KAG-5811

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

File: kong/init.lua [hunk 1]
```diff
@@ -881,23 +881,20 @@ function Kong.init_worker()
...
-        kong.clustering:init_worker()
-        return
+    local is_cp = is_control_plane(kong.configuration)
+    local is_dp_sync_v1 = is_data_plane(kong.configuration) and not kong.sync
+    local using_dedicated = kong.configuration.dedicated_config_processing
 
-      end
+    -- CP needs to support both full and incremental sync
+    -- full sync is only enabled for DP if incremental sync is disabled
+    if is_cp or is_dp_sync_v1 then
```
Comment (chobits) [inline kong/init.lua:890]: In such case, with dedicated_config_processing enabled, the normal workers will also start `clustering:init_worker()`, which is unnecessary.

Comment (chronolaw) [reply kong/init.lua:890]: See https://github.com/Kong/kong/blob/master/kong/clustering/init.lua#L85-L92, DP will only run `init_worker()` in `worker 0` or `privileged agent`

Comment (chobits) [reply kong/init.lua:890]: Referencing [this section in init.lua](https://github.com/Kong/kong/blob/master/kong/clustering/init.lua#L85-L92) and [this section in utils.lua](https://github.com/Kong/kong/blob/master/kong/clustering/utils.lua#L162-L169), DP will only start the clustering process in Worker 0 and the privileged process.

**Review by chobits — APPROVED** [review]

File: kong/init.lua (hunk 1 above)
Comment (chronolaw) [reply kong/init.lua:890]: @StarlightIbuki , do you have some idea about this?

**Review by StarlightIbuki — APPROVED** [review]

**Comment by chronolaw** [comment]
Wait for https://github.com/Kong/kong/pull/13857 cherry-pick successfully.

**Pull request merged**